pkg/module/pdu/pdu.go

```diff
@@ -0,0 +1,259 @@
+
+	return help.String()
+}
+
+const (
+	defaultTimeout = 10 * time.Second // Default timeout for HTTP requests
+	on             = "on"
+	off            = "off"
+	toggle         = "toggle"
+	status         = "status"
+)
+
+func (p *PDU) Init() error {
+	log.Printf("pdu module: Init called - Host: %s, User: %s, Outlet: %d", p.Host, p.User, p.Outlet)
+
+	p.client = &http.Client{Timeout: defaultTimeout}
+
+	controlURL, err := url.Parse(strings.TrimRight(p.Host, "/") + "/control_outlet.htm")
+	if err != nil {
+		return err
+	}
+
+	p.controlURL = controlURL
+
+	statusURL, err := url.Parse(strings.TrimRight(p.Host, "/") + "/status.xml")
+	if err != nil {
+		return err
+	}
+
+	p.statusURL = statusURL
+
+	log.Printf("pdu module: Init completed - controlURL: %s, statusURL: %s", p.controlURL.String(), p.statusURL.String())
+
+	return nil
+}
+
+func (p *PDU) Deinit() error {
+	log.Println("pdu module: Deinit called")
+
+	return nil
+}
+
+func (p *PDU) Run(ctx context.Context, s module.Session, args ...string) error {
+	// Validate outlet number is reasonable (most PDUs have 8-16 outlets max)
+	if p.Outlet < 0 || p.Outlet > 15 {
+		s.Print("Please check the outlet number. It should be normally between 0 and 15. This depends on the PDU model.")
+	}
+
+	if p.client == nil {
+		return fmt.Errorf("PDU client not initialized")
+	}
+
+	if p.Host == "" {
+		return fmt.Errorf("PDU endpoint not configured")
+	}
+
+	if len(args) == 0 {
+		s.Print("No command specified. See 'help' for usage.")
+
+		return nil
+	}
+
+	cmd := strings.ToLower(args[0])
+
+	switch cmd {
+	case on, off, toggle:
+		return p.setPower(ctx, s, cmd)
+	case status:
+		return p.status(ctx, s)
+	default:
+		s.Print("Unknown command: " + cmd)
+		s.Print("Available commands: on, off, toggle, status")
+
+		return nil
+	}
+}
+
+// doRequest creates and executes an HTTP request with authentication and validates the response.
+func (p *PDU) doRequest(ctx context.Context, url string) (*http.Response, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if p.User != "" && p.Password != "" {
+		req.SetBasicAuth(p.User, p.Password)
+	}
+
+	resp, err := p.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		body, _ := io.ReadAll(resp.Body)
+		resp.Body.Close()
+
+		return nil, fmt.Errorf("PDU command failed with status %s: %s", resp.Status, string(body))
+	}
+
+	return resp, nil
+}
+
+func (p *PDU) setPower(ctx context.Context, s module.Session, state string) error {
+	opState, err := parseOp(state)
+	if err != nil {
+		return err
+	}
+
+	q := p.controlURL.Query()
+	q.Set(fmt.Sprintf("outlet%d", p.Outlet), "1")
+	q.Set("op", opState.String())
+	p.controlURL.RawQuery = q.Encode()
+
+	resp, err := p.doRequest(ctx, p.controlURL.String())
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	// Discard response body - we don't need the HTML
+	_, _ = io.Copy(io.Discard, resp.Body)
```

Comment:
Why is it necessary? This seems to do nothing. Don't we need to check for Status code 200 in order to be sure that the request was successful?